code/app/src/main/java/com/example/snackoverflow/AddRecipe.java

```diff
@@ -10,22 +10,60 @@
 import android.net.Uri;
 import android.os.Bundle;
 import android.view.View;
+import android.widget.Button;
+import android.widget.TextView;
 
 import com.github.dhaval2404.imagepicker.ImagePicker;
+import com.google.android.material.textfield.TextInputLayout;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+
 import de.hdodenhof.circleimageview.CircleImageView;
 import kotlin.Unit;
 import kotlin.jvm.functions.Function1;
 
-// TODO: How to request user permission for gallery access with the new Android API
-public class AddRecipe extends AppCompatActivity {
+// TODO: How to request user permission for gallery access with the new
+// Android API
+public class AddRecipe extends AppCompatActivity implements RecipeAddIngredientFragment.OnFragmentInteractionListener{
 
     public CircleImageView imageView;
-
+    private TextInputLayout titletext;
+    private TextInputLayout categorytext;
+    private TextInputLayout servingtext;
```

Comment:
I think it would be better to follow variable naming conventions. For example camel case. titletext -> titleText

Reply:
fixed